src/main/java/io/anserini/index/IndexReaderUtils.java

```diff
@@ -17,23 +17,223 @@
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.apache.lucene.analysis.CharArraySet;
+import org.apache.lucene.analysis.en.EnglishAnalyzer;
 import org.apache.lucene.document.Document;
 import org.apache.lucene.index.IndexableField;
 import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.MultiTerms;
+import org.apache.lucene.index.PostingsEnum;
 import org.apache.lucene.index.Term;
+import org.apache.lucene.index.Terms;
+import org.apache.lucene.index.TermsEnum;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.DocIdSetIterator;
 import org.apache.lucene.search.IndexSearcher;
 import org.apache.lucene.search.Query;
 import org.apache.lucene.search.ScoreDoc;
 import org.apache.lucene.search.TermQuery;
 import org.apache.lucene.search.TopDocs;
 
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+
+import static java.util.stream.Collectors.joining;
 
 public class IndexReaderUtils {
   private static final Logger LOG = LogManager.getLogger(IndexUtils.class);
 
+  public enum DocumentVectorWeight {NONE, TF_IDF}
+
+  public static class NotStoredException extends Exception {
+    public NotStoredException(String message) {
+      super(message);
+    }
+  }
+
+  public static InputStream getReadFileStream(String path) throws IOException {
+    InputStream fin = Files.newInputStream(Paths.get(path), StandardOpenOption.READ);
+    BufferedInputStream in = new BufferedInputStream(fin);
+    if (path.endsWith(".bz2")) {
+      BZip2CompressorInputStream bzIn = new BZip2CompressorInputStream(in);
+      return bzIn;
+    } else if (path.endsWith(".gz")) {
+      GzipCompressorInputStream gzIn = new GzipCompressorInputStream(in);
+      return gzIn;
+    } else if (path.endsWith(".zip")) {
+      GzipCompressorInputStream zipIn = new GzipCompressorInputStream(in);
+      return zipIn;
+    }
+    return in;
+  }
+
+  public static Map<String, String> getTermCounts(IndexReader reader, String termStr) throws IOException, ParseException {
```

Comment:
Let's separate into two methods? `analyzeTerm` to return the stemmed form?
Then, `getTermCounts` can return a`Map<String, Long>`, so we don't need turn numbers into strings...